test/Microsoft.ML.Core.Tests/UnitTests/ColumnTypes.cs

```diff
@@ -0,0 +1,101 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.ImageAnalytics;
+using Xunit;
+namespace Microsoft.ML.Runtime.RunTests
+{
+    public sealed class ColumnTypeTests
+    {
+        [Fact]
+        public void TestEqualAndGetHashCode()
+        {
+            var dict = new Dictionary<ColumnType, string>();
+            // add PrimitiveTypes, KeyType & corresponding VectorTypes
+            PrimitiveType tmp;
+            VectorType tmp1, tmp2;
+            foreach (var kind in (DataKind[])Enum.GetValues(typeof(DataKind)))
+            {
+                tmp = PrimitiveType.FromKind(kind);
+                if(dict.ContainsKey(tmp) && dict[tmp] != tmp.ToString())
```

Comment:
>dict[tmp] != tmp.ToString() [](start = 44, length = 27)

why you need second condition?

Reply:
Because there are duplicate enum values: Num and R4 for ex.

Reply:
Because there are duplicate enum values: Num and R4 for ex.

---
In reply to: [208363396](https://github.com/dotnet/machinelearning/pull/620#discussion_r208363396) [](ancestors = 208363396)

Reply:
Well, why you do it for ImageTypes then?

---
In reply to: [208401357](https://github.com/dotnet/machinelearning/pull/620#discussion_r208401357) [](ancestors = 208401357,208363396)

Reply:
oh, common, for uniformity sake :). I can remove this check for ImageType only

---
In reply to: [208416968](https://github.com/dotnet/machinelearning/pull/620#discussion_r208416968) [](ancestors = 208416968,208401357,208363396)